keps/sig-node/5067-pod-generation/README.md

```diff
@@ -0,0 +1,1066 @@
+generally found in any type of pod update.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+* Provide a general solution for the pod status to express which pod update is 
+currently being reflected.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+* Expand the set of mutable fields.
+* Populate `metadata.generation` and `status.ObservedGeneration` on mirror pods.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### Current behavior
+
+The pod [`metadata.generation`](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#metadata) field does exist today, and is 
+documented as "a sequence number representing a specific generation of the 
+desired state. Set by the system and monotonically increasing, per-resource." 
+
+Its current behavior in pods is:
+
+* Pod `metadata.generation` is not populated by default by the system.
+* The client can custom-set the `metadata.generation` on pod create.
+* `metadata.generation` cannot be updated by the client.
+* `metadata.generation` does not get incremented by the system when the podspec is updated.
+* `metadata.generation` does get incremented by the system when `DeletionTimestamp` is set.
+
+### API Changes
+
+#### Generation
+
+The `metadata.generation` field is currently unused on pods, but we can start 
+leveraging it to help track which pod state is currently being reflected in the 
+pod status. For consistency, pod 
+`metadata.generation` will be incremented whenever the pod has changes that the kubelet
+needs to actuate. 
+
+#### ObservedGeneration
+
+A new optional field `status.observedGeneration` field will be added to the pod status.
+Kubelet will set this to communicate which pod state is being expressed in the 
+current pod status. This is analogous to the `status.observedGeneration` that exists
+in other resources' statuses such as [StatefulSets](https://github.com/kubernetes/api/blob/9e7d345b161c12a3056efa88e285b3ef68450c54/apps/v1/types.go#L274).
+
+The `status.observedGeneration` may not necessarily be a reflection of every single
+field in the pod status. Instead, it reports the latest generation that the kubelet
+has seen. This means that `status.observedGeneration` captures the kubelet's
+decision to admit a change, and acknowledge that it has seen the pod's new spec.
+It will not necessarily be able to reflect that the kubelet has completed actuation
```

Comment:
```suggestion
There are cases when `status.observedGeneration` may be behind - other status values already reflect a next generation, but the next update from kubelet SHOULD bring the  `status.observedGeneration` to the current value.
It also will not necessarily be able to reflect that the kubelet has completed actuation
```

Reply:
it may be important note for the cases when there are a lot of rapid spec changes and kubelet is constantly behind.

Reply:
thanks! added to the kep